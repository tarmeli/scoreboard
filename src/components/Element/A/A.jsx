import React from "react";
import Element from "../Element";

const A = ({ className, children, ...props }) => (
    <Element tagName="a" className={className} {...props}>
        {children}
    </Element>
);

export default A;
