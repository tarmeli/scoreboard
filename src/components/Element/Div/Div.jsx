import React from "react";
import Element from "../Element";

const Div = ({ className, children, ...props }) => (
    <Element tagName="div" className={className} {...props}>
        {children}
    </Element>
);

export default Div;
