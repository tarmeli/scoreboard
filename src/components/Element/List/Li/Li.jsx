import React from "react";
import Element from "../../Element";

const Li = ({ className, children, ...props }) => (
    <Element tagName="li" className={className} {...props}>
        {children}
    </Element>
);

export default Li;
