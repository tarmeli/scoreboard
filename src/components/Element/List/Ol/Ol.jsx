import React from "react";
import Element from "../../Element";

const Ol = ({ className, children, ...props }) => (
    <Element tagName="ol" className={className} {...props}>
        {children}
    </Element>
);

export default Ol;
