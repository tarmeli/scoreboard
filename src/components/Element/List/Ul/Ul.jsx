import React from "react";
import Element from "../../Element";

const Ul = ({ className, children, ...props }) => (
    <Element tagName="ul" className={className} {...props}>
        {children}
    </Element>
);

export default Ul;
