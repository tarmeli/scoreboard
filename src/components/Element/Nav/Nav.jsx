import React from "react";
import Element from "../Element";

const Nav = ({ className, children, ...props }) => (
    <Element tagName="nav" className={className} {...props}>
        {children}
    </Element>
);

export default Nav;
