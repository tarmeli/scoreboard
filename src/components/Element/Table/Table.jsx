import React from "react";
import Element from "../Element";

const Table = ({ className, children, ...props }) => (
    <Element tagName="table" className={className} {...props}>
        {children}
    </Element>
);

export default Table;
