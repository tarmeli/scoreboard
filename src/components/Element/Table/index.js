import Tbody from "./Tbody/Tbody";
import Td from "./Td/Td";
import Th from "./Th/Th";
import Thead from "./Thead/Thead";
import Tr from "./Tr/Tr";

export { Tbody, Td, Th, Thead, Tr };
