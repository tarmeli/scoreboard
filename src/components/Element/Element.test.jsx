import React from "react";
import { shallow } from "enzyme";

import Div from "./Div/Div";
import A from "./A/A";
import Nav from "./Nav/Nav";
import Ol from "./List/Ol/Ol";
import Li from "./List/Li/Li";
import Ul from "./List/Ul/Ul";
import Table from "./Table/Table";
import Td from "./Table/Td/Td";
import Tr from "./Table/Tr/Tr";
import Th from "./Table/Th/Th";
import Tbody from "./Table/Tbody/Tbody";
import Thead from "./Table/Thead/Thead";

const availableTags = [
    A,
    Div,
    Nav,
    Ol,
    Li,
    Ul,
    Table,
    Td,
    Tr,
    Th,
    Tbody,
    Thead,
];

describe("Element", () => {
    let component;

    availableTags.forEach((Tag) => {
        it(`can render "${Tag.name}"`, () => {
            component = shallow(<Tag />);

            expect(component.render()).toMatchSnapshot();
        });
    });
});
