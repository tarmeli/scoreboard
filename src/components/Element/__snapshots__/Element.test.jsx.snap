// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element can render "A" 1`] = `<a />`;

exports[`Element can render "Div" 1`] = `<div />`;

exports[`Element can render "Li" 1`] = `<li />`;

exports[`Element can render "Nav" 1`] = `<nav />`;

exports[`Element can render "Ol" 1`] = `<ol />`;

exports[`Element can render "Table" 1`] = `<table />`;

exports[`Element can render "Tbody" 1`] = `<tbody />`;

exports[`Element can render "Td" 1`] = `<td />`;

exports[`Element can render "Th" 1`] = `<th />`;

exports[`Element can render "Thead" 1`] = `<thead />`;

exports[`Element can render "Tr" 1`] = `<tr />`;

exports[`Element can render "Ul" 1`] = `<ul />`;
